ecl/eclcc/WORKUNITS.rst

```diff
@@ -0,0 +1,862 @@
+=======================
+Understanding workunits
+=======================
+
+Introduction
+************
+
+A workunit contains all the information that the system requires about a query - including what parameters it takes,
+how to execute it, and how to format the results.  Understanding the contents of a workunit is a key step to
+understanding how the HPCC system fits together.  This document begins with an overview of the different elements
+in a workunit.  That is followed by a walk-through executing a simple query, with a more detailed description of
+some of the workunit components, to show how they all tie together.
+
+Before looking at the contents of the workunit it is important to understand one of the design goals behind the HPCC
+system.  The ECL system logically splits that code that is needed to execute a query in two.  One the one hand there
+are the algorithms that are used a perform different dataset operations (e.g., sorting, deduping).  The same algorithms are
+used by all the queries that execute on the system.  On the other hand there is the meta-data describing what columns
+are in the datasets, which columns you need to sort by, and the order of operations required by the query.  These
+are typically different for each query.
+
+The workunit only contains information about the meta data i.e., "what to do", while the different engines (hthor,
+roxie, thor) implement the algorithms - "how to do it".  If you look at a workunit for an ECL query that sorts
+a dataset you will not find code to perform the sort in the workunit - you won't even find a call to a sort library
+function - that logical is contained in the engine that executes the query.
+
+One consequence of this split, which can be initially confusing, is that execution continually passes back and forth
+between the generated code and the engines.  By the end of this document you should have a better understanding
+of how the generated code is structured and what part it plays in executing a query.
+
+Note the term "Query" is used as a generic term to cover read-only queries (typically run in roxie) and ETL style
+queries which create lots of persistent datafiles (typically run in Thor).  Also, the term "workunit" is used a bit
+ambiguously.  The dll created from a query is called a workunit (which is static), but "workunit" is also used to
```

Comment:
The ambiguity here will seem very odd! *We* make these references as a basis of 'history' however, this point is not clear. Instead the ambiguity lies within this documented when it could have been easily disambiguated when written.

Perhaps this an an opportunity to officially disambiguate?
If not, I would add "... is used a bit ambiguously amongst the HPCC-Systems community"